server/job/plexsync/index.ts

```diff
@@ -351,6 +462,18 @@ class JobPlexSync {
     logger[level](message, { label: 'Plex Sync', ...optional });
   }
 
+  // checks if any of this.libraries has Hama agent set in Plex
+  private async hasHamaAgent() {
+    const plexLibraries = await this.plexClient.getLibraries();
+    for (const library of this.libraries) {
+      const found = plexLibraries.find((lib) => lib.key == library.id);
+      if (found && found.agent == HAMA_AGENT) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
Strict comparison and some optional chaining
```suggestion
    const plexLibraries = await this.plexClient.getLibraries();
    for (const library of this.libraries) {
      const found = plexLibraries.find((lib) => lib.key === library.id);
      if (found?.agent === HAMA_AGENT) {
        return true;
      }
    }
    return false;
```

I would probably write this code like this though. (Wrote this in GitHub so may not be syntactically correct)
```suggestion
    const plexLibraries = await this.plexClient.getLibraries();
    return this.libraries.some((library) => plexLibraries.some((plexLibrary) => plexLibrary.agent === HAMA_AGENT && library.id === plexLibrary.key));
```